DB/src/main/java/io/deephaven/db/v2/by/StreamFirstByChunkedOperator.java

```diff
@@ -0,0 +1,237 @@
+        inputColumns = new ColumnSource[numResultColumns];
+        outputColumns = new WritableSource[numResultColumns];
+        final Map<String, ArrayBackedColumnSource<?>> resultColumnsMutable = new LinkedHashMap<>(numResultColumns);
+        for (int ci = 0; ci < numResultColumns; ++ci) {
+            final MatchPair resultPair = resultPairs[ci];
+            final ColumnSource<?> streamSource = streamTable.getColumnSource(resultPair.left());
+            final ArrayBackedColumnSource<?> resultSource = ArrayBackedColumnSource.getMemoryColumnSource(0, streamSource.getType(), streamSource.getComponentType());
+            resultColumnsMutable.put(resultPair.left(), resultSource);
+            inputColumns[ci] = ReinterpretUtilities.maybeConvertToPrimitive(streamSource);
+            outputColumns[ci] = (WritableSource<?>) ReinterpretUtilities.maybeConvertToPrimitive(resultSource);
+            Assert.eq(inputColumns[ci].getChunkType(), "inputColumns[ci].getChunkType()", outputColumns[ci].getChunkType(), "outputColumns[ci].getChunkType()");
+        }
+        resultColumns = Collections.unmodifiableMap(resultColumnsMutable);
+        cachedRedirections = new SoftReference<>(redirections = new LongArraySource());
+    }
+
+    @Override
+    public void ensureCapacity(final long tableSize) {
+        redirections.ensureCapacity(tableSize - previousTableSize);
+    }
+
+    @Override
+    public Map<String, ? extends ColumnSource<?>> getResultColumns() {
+        return resultColumns;
+    }
+
+    @Override
+    public void startTrackingPrevValues() {
+        Arrays.stream(outputColumns).forEach(ColumnSource::startTrackingPrevValues);
+    }
+
+    @Override
+    public boolean requiresIndices() {
+        return true;
+    }
+
+    @Override
+    public boolean unchunkedIndex() {
+        return true;
+    }
+
+    @Override
+    public void resetForStep(@NotNull final ShiftAwareListener.Update upstream) {
+        if ((redirections = cachedRedirections.get()) == null) {
+            cachedRedirections = new SoftReference<>(redirections = new LongArraySource());
+        }
+    }
+
+    @Override
+    public void addChunk(final BucketedContext context, // Unused
+                         final Chunk<? extends Values> values, // Unused
+                         @NotNull final LongChunk<? extends KeyIndices> inputIndices,
+                         @NotNull final IntChunk<KeyIndices> destinations,
+                         @NotNull final IntChunk<ChunkPositions> startPositions,
+                         final IntChunk<ChunkLengths> length, // Unused
+                         @NotNull final WritableBooleanChunk<Values> stateModified) {
+        for (int ii = 0; ii < startPositions.size(); ++ii) {
+            final int startPosition = startPositions.get(ii);
+            final long destination = destinations.get(startPosition);
+            redirections.set(destination - previousTableSize, inputIndices.get(startPosition));
```

Comment:
How are you checking that this isn't just an additon to a previous state and should be ignored?

Reply:
Oops, I wasn't. Fixed in next commit.